frontend/apps/cloudstorage/webdavapi.lua

```diff
@@ -133,7 +133,7 @@ function WebDavApi:listFolder(address, user, pass, folder_path, folder_mode)
                 item:find("<[^:]*:resourcetype></[^:]*:resourcetype>")
 
             local item_path = (path == "" and has_trailing_slash) and item_name or path .. "/" .. item_name
-            if item:find("<[^:]*:collection/>") then
+            if item:find("<[^:]*:collection[^<]*/>") then
```

Comment:
Not really interested with webdav, but I stopped a few minutes on this regexp.
Doing tests, it seems ok.
But I'm more used to seeing `[^Z]*Z` , that is, anything but Z until meeting Z - than anything but Y until meeting Z.
So, `collection[^/]*/>"` wouldn't have stopped me, and may just work?

Reply:
Thanks for reviewing. 

The matching in the webdav code is a bit of a hack anyway, it's not proper xml parsing. The original regexp and the proposed one only match a self-closing element, not an element with a value or nested elements inside.

My thought was for the regexp to prevent a match on the self-closing `/>` of an unrelated element by including `[^<]` to match any character but the start of another element. Matches are sometimes done greedy (match as long a string as possible). Contrived example of an unwanted positive match I was trying to prevent:

```
<d:collection xmlns:d="DAV:">stranger things</d:collection><SomeOtherElement/>
^^^^^^^^^^^^^                                                               ^^
```

Yours might fail on the slash in (again theoretical):

`
<d:collection** xmlns:d="DAV:" path="downloads/books"/>
`

Both will do the job just fine for my real-life example though.

Reply:
> Yours might fail on the slash in (again theoretical):

Alright :) I see there is some thoughts in there, so it's not random :)
Good for me !